lib/v2/dlnews/category.js

```diff
@@ -0,0 +1,109 @@
+    const list = getList(data);
+    const items = await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: response } = await got(item.link);
+                const $ = cheerio.load(response);
+                const scriptTagContent = $('script#fusion-metadata').text();
+                const jsonData = JSON.parse(scriptTagContent.split('Fusion.globalContent=')[1].split('Fusion.globalContentConfig')[0].replace(/(;)/g, '')).content_elements;
+                let end = false;
+                item.description = jsonData.reduce((acc, curr) => {
+                    let data = '';
+                    if (!end) {
+                        switch (curr.type) {
+                            case 'custom_embed':
+                                if (curr.embed.config.text) {
+                                    data = '<ul>';
+                                    data =
+                                        data +
+                                        curr.embed.config.text
+                                            .split('\n')
+                                            .filter(Boolean)
+                                            .map((v) => `<li>${v}</li>`)
+                                            .join('');
+                                    data = data + '</ul>';
+                                }
+
+                                break;
+                            case 'header':
+                                data = `<h2>${curr.content}</h2>`;
+                                if (data.includes('What we’re reading')) {
+                                    data = '';
+                                    end = true;
+                                }
+                                break;
+                            case 'list':
+                                data = curr.items.map((v) => `<li>${v.content}</li>`).join('');
+                                data = curr.list_type === 'unordered' ? `<ul>${data}</ul>` : `<ol>${data}</ol>`;
+                                break;
+                            case 'image':
+                                data = '<figure>';
+                                data = data + `<img src=${curr.url} alt=${curr.alt_text}/>`;
+                                data = data + `<figcaption> ${curr.subtitle}</figcaption>`;
+                                data = data + '</figure>';
+                                break;
+                            case 'text':
+                                data = curr.content;
+                                if (data.includes('<b>NOW READ: </b>')) {
+                                    data = '';
+                                }
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+
+                    return acc + data;
+                }, '');
```

Comment:
Please use art-template. docs: https://docs.rsshub.app/en/joinus/advanced/script-standard#rendering-templates
The template will look somehow like
```handlebars
{{ each data d }}
    {{ if d.type === 'custom_embed' }}
        <ul>
            {{ each d.embed.config.text.split('\n') line }}
                <li>{{ line }}</li>
            {{ /each }}
        </ul>
    {{ else if d.type === 'oembed_response' }}
        {{@ d.raw_oembed.html }}
    {{ else if d.type === 'header' }}
        <h{{ d.level }}>{{ d.content }}</h{{ d.level }}>
    {{ else if d.type === 'image' }}
        <figure>
            <img src="{{ d.url }}" alt="{{ d.alt_text }}">
            <figcaption>{{ d.subtitle }}</figcaption>
        </figure>
    {{ else if d.type === 'list' }}
        {{ if d.list_type === 'unordered' }}<ul>{{ else }}<ol>{{ /if }}
            {{ each d.items item }}
                <li>{{@ item.content }}</li>
            {{ /each }}
        {{ if d.list_type === 'unordered' }}</ul>{{ else }}</ol>{{ /if }}
    {{ else if d.type === 'text' }}
        <p>{{@ d.content }}</p>
    {{ /if }}
{{ /each }}

```